internal/impl/deltalake/output.go

```diff
@@ -0,0 +1,202 @@
+package deltalake
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/redpanda-data/benthos/v4/public/service"
+	"github.com/rivian/delta-go"
+	"github.com/rivian/delta-go/lock/nillock"
+	"github.com/rivian/delta-go/state"
+	"github.com/rivian/delta-go/storage"
+	"github.com/rivian/delta-go/storage/s3store"
+)
+
+const (
+	dlFieldTableURL = "table_url"
+	dlFieldBatching = "batching"
+)
+
+func outputConfigSpec() *service.ConfigSpec {
+	return service.NewConfigSpec().
+		Beta().
+		Categories("Services").
+		Version("4.53.0").
+		Summary("Ingests data into Delta Lake tables such as those managed by Databricks").
+		Description(`
+    Ingests parquet data into Delta Lake tables, supporting both Databricks-managed and open-source Spark deployments.
+`).
+		Fields(
+			service.NewStringField(dlFieldTableURL).
+				Description("The URL to table to ingest data into. Infers the storage backend to use from the scheme in the given table URL."),
+			service.NewOutputMaxInFlightField(),
+		)
+}
+
+func init() {
+	err := service.RegisterOutput("delta_lake", outputConfigSpec(),
+		func(conf *service.ParsedConfig, mgr *service.Resources) (out service.Output, maxInFlight int, err error) {
+			if maxInFlight, err = conf.FieldMaxInFlight(); err != nil {
+				return
+			}
+			out, err = newDeltaLakeWriterFromParsed(conf, mgr)
+			return
+		})
+	if err != nil {
+		panic(err)
+	}
+}
+
+func newDeltaLakeWriterFromParsed(conf *service.ParsedConfig, mgr *service.Resources) (*deltaLakeWriter, error) {
+	d := &deltaLakeWriter{logger: mgr.Logger()}
```

Comment:
It would be easier to read if deltaLakeWriter was instantiated by the end of the function.
We'd also avoid `var err error`.